sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -90,56 +88,69 @@ object SerializedHostTableUtils {
   }
 }
 
+/**
+ * Class that is used to broadcast results (a contiguous host batch) to executors.
+ *
+ * This is instantiated in the driver, serialized to an output stream provided by Spark
+ * to broadcast, and deserialized on the executor. Both the driver's and executor's copies
+ * are cleaned via GC. Because Spark closes `AutoCloseable` broadcast results after spilling
+ * to disk, this class does not subclass `AutoCloseable`. Instead we implement a `closeInternal`
+ * method only to be triggered via GC.
+ *
+ * @param data HostConcatResult populated for a broadcast that has column, otherwise it is null.
+ *             It is transient because we want the executor to deserialize its `data` from Spark's
+ *             torrent-backed input stream.
+ * @param output used to find the schema for this broadcast batch
+ * @param numRows number of rows for this broadcast batch
+ * @param dataLen size in bytes for this broadcast batch
+ */
 // scalastyle:off no.finalize
 @SerialVersionUID(100L)
 class SerializeConcatHostBuffersDeserializeBatch(
-    data: Array[SerializeBatchDeserializeHostBuffer],
-    output: Seq[Attribute])
-  extends Serializable with AutoCloseable with Logging {
+    @transient var data: HostConcatResult,
+    output: Seq[Attribute],
+    var numRows: Int,
+    var dataLen: Long)
+  extends Serializable with Logging {
   @transient private var dataTypes = output.map(_.dataType).toArray
-  @transient private var headers = data.map(_.header)
-  @transient private var buffers = data.map(_.buffer)
 
   // used for memoization of deserialization to GPU on Executor
   @transient private var batchInternal: SpillableColumnarBatch = null
 
+  private def maybeGpuBatch: Option[SpillableColumnarBatch] = Option(batchInternal)
+
```

Comment:
Because of `Serializable`.

On deserialization, a `batchInternal: Option[SpillableColumnarBatch]` will not be `None`, it will be `Null`. So, I thought this might be a bit cleaner.